controller/Makefile

```diff
@@ -89,7 +89,7 @@ KIND ?= kind
 ##@ Build Dependencies
 
 ## Location to install dependencies to
-LOCALBIN ?= $(shell pwd)/bin
+LOCALBIN ?= $(shell cd .. && pwd)/bin
```

Comment:
What about using
```
LOCALBIN ?= $(dir $(lastword $(MAKEFILE_LIST)))bin
```

in the common.mk?

Reply:
@Chever-John 
Would you give this a try? Thanks!

Reply:
Sry for late~

Yes, I have tried it.

I added the following code (A) into the common.mk file and commented out the corresponding code in the controller/Makefile as shown below (B):

A. code
```makefile
LOCALBIN ?= $(dir $(lastword $(MAKEFILE_LIST)))
$(LOCALBIN):
	@mkdir -p $(LOCALBIN)
```

B. code

comment out the `LOCALBIN ?= $(shell cd .. && pwd)/bin`

```makefile
## Location to install dependencies to
#LOCALBIN ?= $(shell cd .. && pwd)/bin
# Remember to remove tools downloaded into bin directory manually before updating them.
# If they need to be updated frequently, we can consider to store them in the `Dockerfile.dev`.
$(LOCALBIN):
	mkdir -p $(LOCALBIN)
```

And I tried to run the command `make sth` in the `controller` dir as follows:

```shell
╭─cheverjohn at Jonathan’s MacBook Pro in ~/Workspace/OpenSource/github.com/Chever-John/htnngit:(chore/unite-the-bin-cache-dir*)
╰─± cd controller 
╭─cheverjohn at Jonathan’s MacBook Pro in ~/Workspace/OpenSource/github.com/Chever-John/htnn/controllergit:(chore/unite-the-bin-cache-dir*)
╰─± make controller-gen
test -s ..//controller-gen && ..//controller-gen --version | grep -q v0.13.0 || \
        GOBIN=../ go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.13.0
cannot install, GOBIN must be an absolute path
make: *** [..//controller-gen] Error 1
```

Then errors came out:)

Reply:
How about use the following code:

```makefile
# common.mk
ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
LOCALBIN := $(ROOT_DIR)/bin
$(LOCALBIN):
	@mkdir -p $(LOCALBIN)
```

Reply:
@Chever-John 
LGTM

Reply:
Update!!!

from `ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))` to `ROOT_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))`